clang/test/CIR/Lowering/ThroughMLIR/if.c

```diff
@@ -0,0 +1,137 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -fno-clangir-direct-lowering -emit-mlir %s -o %t.mlir
+// RUN: FileCheck --input-file=%t.mlir %s
+
+void foo() {
+  int a = 2;
+  int b = 0;
+  if (a > 0) {
+    b++;
+  } else {
+    b--;
+  }
+}
+
+//CHECK: func.func @foo() {
+//CHECK:   %alloca = memref.alloca() {alignment = 4 : i64} : memref<i32>
```

Comment:
Please use regex for the check lines in this file, you can find examples elsewhere (see clang/test/CIR/CodeGen/abstract-cond.c and others)